### ISSUE TITLE
AL03: Add option to skip alias requirement for scalar subqueries in WHERE/HAVING

### ISSUE THREAD

**Issue by grmaxwell** [sqlfluff/sqlfluff#7653]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### Description

AL03 currently flags expressions without aliases inside scalar subqueries used for comparison:

```
SELECT * FROM t WHERE x > (SELECT MAX(y) FROM t2)
-- AL03 flags MAX(y) as needing an alias
```
This is unnecessary because:

- The subquery returns a single scalar value for comparison, not a result column
- Adding an alias (`MAX(y) AS m`) has no effect on the query output or readability
- It's a common pattern in incremental dbt models

**Proposed solution:** Add `allow_scalar_subquery` config option (default: false for backwards compatibility) that skips the check when the expression is inside a subquery used in WHERE/HAVING context.

### Use case

A way to tell AL03 that scalar subqueries used in WHERE/HAVING comparisons don't need aliases, while still enforcing aliases for expressions in SELECT lists.

### Dialect

All / Not dialect-specific This is standard SQL behavior - scalar subqueries in WHERE/HAVING work identically across all supported dialects.

### Are you willing to work on and submit a PR to address the issue?

- [x] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by peterbud** [comment]
Hi @grmaxwell - thanks for opening this issue.

I'm a bit confused for three reasons. 

1. The above query does NOT trigger AL02, but RF02 rule:
```bash
sqlfluff lint test.sql -d ansi
== [test.sql] FAIL                                                                                                
L:   1 | P:   1 | AM04 | Query produces an unknown number of result columns.
                       | [ambiguous.column_count]
L:   1 | P:  17 | LT14 | The 'WHERE' keyword should always start a new line.
                       | [layout.keyword_newline]
L:   1 | P:  39 | RF02 | Unqualified reference 'y' found in select with more than
                       | one referenced table/view.
                       | [references.qualification]
L:   2 | P:   1 | LT12 | Files must end with a single trailing newline.
                       | [layout.end_of_file]
All Finished 📜 🎉!
```

2. `AL03` rule does have an `allow_scalar` setting already, see the rule [reference](https://docs.sqlfluff.com/en/stable/reference/rules.html#rule-AL03)

3. The `RF02` rule also have a setting `subqueries_ignore_external_references` - if you turn this one the above mentioned linting warning goes away.

Can you clarify these pls? Maybe a better example which triggers `AL03` for you?

**Comment by peterbud** [comment]
Closing due to lack of response
